### ISSUE TITLE
`poetry env use` and `poetry env remove` are based on the directory name instead of `tool.poetry.name`

### ISSUE THREAD

**Issue by maggyero** [sdispater/poetry#1425]
The `poetry env use {python}` and `poetry env remove {python}` commands incorrectly use the directory name of a project instead of the `tool.poetry.name` property of the `pyproject.toml` file to create/delete a virtual environment.

This clashes with the `poetry add`, `poetry build`, `poetry install`, `poetry lock`, `poetry remove`, `poetry run`, `poetry shell`, `poetry show` and `poetry update` commands which correctly use the `tool.poetry.name` property of the `pyproject.toml` file to implicitly create a virtual environment.

## Example

For instance if a virtual environment was implicitly created with `poetry install` (so based on `tool.poetry.name`):

```
$ poetry install
Creating virtualenv rule_detection-Yz5oMgHf-py3.7 in C:\Users\maggyero\AppData\Local\pypoetry\Cache\virtualenvs
Using virtualenv: C:\Users\maggyero\AppData\Local\pypoetry\Cache\virtualenvs\{tool.poetry.name}-Yz5oMgHf-py3.7
```

and then removed with `poetry env remove python` (so based on the directory name of the project), a `ValueError` is raised if `{directory_name}` ≠ `{tool.poetry.name}`:

```
$ poetry env remove python
[ValueError]
Environment "{directory_name}-Yz5oMgHf-py3.7" does not exist.
```

FYI @stephsamson @sdispater

**Comment by KelSolaar** [comment]
Hi @maggyero,

I have referenced it here: #1417

**Comment by sdispater** [comment]
This has been fixed in #1477

**Comment by maggyero** [comment]
Thanks a lot @sdispater!
